modules/ROOT/pages/dataweave-formats-binary.adoc

```diff
@@ -1,42 +1,55 @@
-= Binary Format
 
+= Binary Format
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: format, binary, application/octet-stream, application/x-binary, application/octet-stream
 
-MIME Type: `application/octet-stream`
+MIME type: `application/octet-stream`
 
 ID: `binary`
 
 The Binary data format handles binary content, such as an image or PDF. Such content is represented as a `Binary` type.
 
 Note that Binary content is stored into RAM memory.
 
+// TODO: NEED EXAMPLES AND/OR MORE INFO ON BINARY FORMAT
+
+
+// CONFIG PROPS GET APPENDED BELOW /////////////////////////////////
 
-// CONFIG PROPS ///////////////////////////////////////////////////////
 
 [[properties]]
 == Configuration Properties
 
 DataWeave supports the following configuration properties for this format.
 
+
+[[reader_properties]]
 === Reader Properties
 
-There are no reader properties for Binary data.
+There are no reader properties for this format.
 
+ 
+[[writer_properties]]
 === Writer Properties
 
-The Binary format accepts properties that provide instructions for writing output data.
+This format accepts properties that provide instructions for writing output data.
 
 [cols="1,1,1,3a", options="header"]
 |===
-| Parameter | Type | Default | Description
-| `bufferSize` | `Number` | `8192` | Size of the writer buffer.
-| `deferred` | `Boolean` | `false` | When set to `true`, DataWeave generates the output as a data stream, and the script's execution is deferred until it is consumed.
-  Valid values are `true` or `false`.
+|Parameter |Type |Default|Description
+|`bufferSize`  |`Number`|`8192`|Size of the buffer writer.
+|`deferred`  |`Boolean`|`false`|Generates the output as a data stream when set to `true`, and defers the script's execution until consumed.
```

Comment:
TODO: until what is consumed?

Reply:
until the generated content is consumed, in other words it will execute the transformation when the content is actually used (for example in a subsequent message processor)

Reply:
```suggestion
|`deferred`  |`Boolean`|`false`|Generates the output as a data stream when set to `true`, and defers the script's execution until the generated content is consumed.
```